src/sortStudents.ts

```diff
@@ -1,16 +1,96 @@
 
 export interface Student {
   // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'Name',
+  Surname = 'Surname',
+  Age = 'Age',
+  Married = 'Married',
+  AverageGrade = 'AverageGrade',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
-
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
   // write your function
+  let sortCallback: (firstStudent: Student, secondStudent: Student) => number;
+  let sortOrder: number;
+
+  switch (order) {
+    case 'asc':
+      sortOrder = 1;
+      break;
+    case 'desc':
+      sortOrder = -1;
+      break;
+    default:
+      return [];
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+      sortCallback = (
+        firstStudent: Student,
+        secondStudent: Student,
+      ): number => firstStudent
+        .name
+        .localeCompare(secondStudent.name)
+        * sortOrder;
+      break;
+    case SortType.Surname:
+      sortCallback = (
+        firstStudent: Student,
+        secondStudent: Student,
+      ): number => firstStudent
+        .surname
+        .localeCompare(secondStudent.surname)
+        * sortOrder;
+      break;
```

Comment:
you can do this using only 1 case like 
```suggestion
case SortType.Name:
case SortType.Surname:
```

think how to make it and, do this also with next cases